src/sidebar/services/EmbedderAnalyticsService.ts

```diff
@@ -0,0 +1,27 @@
+import type { SidebarSettings } from '../../types/config';
+import * as postMessageJsonRpc from '../util/postmessage-json-rpc';
+
+/**
+ * @inject
+ */
+export class EmbedderAnalyticsService {
```

Comment:
I considered reusing the `AnalyticsService` we introduced in https://github.com/hypothesis/client/pull/6275, but it feels like something fundamentally different, so I ended-up creating a new service.